exercises/nth-prime/src/lib.rs

```diff
@@ -0,0 +1,3 @@
+pub fn nth(n: u32) -> Result<u32, ()> {
```

Comment:
WRT the error type of the result, I'm of two minds.

On the one hand, `Result<T, ()>` is bad style: if you can get away with an empty tuple as your error type, you should be using `Option<T>` instead. 

On the other hand, it's legitimate to want to familiarize students with `Result`, even if we actually don't care what error type they use.

On the gripping hand, this exercise is number 6 of currently 80 exercises in Rust, per

```bash
$ jq '{slug: .exercises[].slug}' config.json | grep slug | nl | grep nth-prime
     6    "slug": "nth-prime"
```

Students attempting this exercise are still expected to be in the early stages of Rust learning. This implies that we _do_ want to provide a stub for them, and _do not_ want them to need to deal with traits, generics, or other advanced features. 

I think the signature I'd like to see looks like this:

```rust
pub fn nth(_: usize) -> Result<usize, &'static str> {
    unimplemented!()
}
```

(`'static` is a special lifetime for things which are known to last the entire lifetime of the program, such as string literals.) This still allows the `Err("Some message")`  usage without requiring people to write a generic lifetime just yet.

However, I'm not 100% convinced of my own correctness yet in this case, and would like to invite comments from other maintainers about this proposal.

Reply:
I agree with not showing `Result<T, ()>` early.

Now, it does lead me to suggest a possible option: Let's just change this execise to use `Option` instead of `Result`. Don't need `Result` yet.

`Result<usize, &'static str>` is OK to start. I'll attempt to predict some possible objections to them and respond:

* students may not know what `&'static` is.
  * Okay, then maybe we can explain it like we did in https://github.com/exercism/rust/blob/master/exercises/hello-world/src/lib.rs#L1
* it's bad style to use a string for an error; instead it should be a type that the caller can inspect to determine the cause of the error
  * Okay, maybe we can set the example to use a `NoSuchPrime` type, that's fine. Should the test suite explicitly check for that type, OR should it just look for `is_err`?

Reply:
I'd be fine with just using `Option<T>`. That broadens the scope of this PR, but in a way that I'd be willing to accept. 

@cbzehner would you be willing to rework the exercise such that tests expect an `Option`?

Reply:
Yes, I'd be happy to.